Sources/Components/New Group/InlineConsentView.swift

```diff
@@ -0,0 +1,126 @@
+        label.numberOfLines = 0
+        return label
+    }()
+
+    private lazy var yesButton: Button = {
+        let button = Button(style: .callToAction)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addTarget(self, action: #selector(yesButtonTapped), for: .touchUpInside)
+        return button
+    }()
+
+    private lazy var infoButton: Button = {
+        let button = Button(style: .flat)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addTarget(self, action: #selector(infoButtonTapped), for: .touchUpInside)
+        return button
+    }()
+
+    // MARK: - External properties / Dependency injection
+
+    public weak var delegate: InlineConsentViewDelegate?
+
+    public var descriptionText: String = "" {
+        didSet {
+            descriptionTitleLabel.text = descriptionText
+        }
+    }
+
+    public var yesButtonTitle: String = "" {
+        didSet {
+            yesButton.setTitle(yesButtonTitle, for: .normal)
+        }
+    }
+
+    public var infoButtonTitle: String = "" {
+        didSet {
+            infoButton.setTitle(infoButtonTitle, for: .normal)
+        }
+    }
+
+    // MARK: - Setup
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+
+    private func setup() {
+        addSubview(descriptionTitleLabel)
+        addSubview(yesButton)
+        addSubview(infoButton)
+
+        NSLayoutConstraint.activate([
+            descriptionTitleLabel.topAnchor.constraint(equalTo: topAnchor),
+            descriptionTitleLabel.leadingAnchor.constraint(equalTo: leadingAnchor),
+            descriptionTitleLabel.trailingAnchor.constraint(equalTo: trailingAnchor),
+
+            yesButton.topAnchor.constraint(equalTo: descriptionTitleLabel.bottomAnchor, constant: .mediumLargeSpacing),
+            yesButton.centerXAnchor.constraint(equalTo: centerXAnchor),
+
+            infoButton.topAnchor.constraint(equalTo: yesButton.bottomAnchor, constant: .mediumLargeSpacing),
+            infoButton.centerXAnchor.constraint(equalTo: centerXAnchor),
+            infoButton.bottomAnchor.constraint(equalTo: bottomAnchor),
+        ])
+    }
+
+    // MARK: - Actions
+
+    @objc private func yesButtonTapped(_ sender: Button) {
+        delegate?.inlineConsentView(self, didSelectActivateButton: sender)
+    }
+
+    @objc private func infoButtonTapped(_ sender: Button) {
+        delegate?.inlineConsentView(self, didSelectInfoButton: sender)
+    }
+
+    // MARK: - Superclass overrides
+
+    public override var intrinsicContentSize: CGSize {
+        let intermediateSpacing = CGFloat.mediumLargeSpacing * 2
+
+        guard let descriptionText = descriptionTitleLabel.text else {
+            return CGSize.zero
+        }
+        let screenSize = UIScreen.main.bounds
```

Comment:
I'm a little uncomfortable with using the `UIScreen.main` singleton to get the screen size here.🤔 Especially since this is a view created in a framework. Would it be better to ask the`delegate` to provide the screen size?

Reply:
I'm still having a hard time understanding why this is needed, feels like to goes against what Auto Layout stands for, we even have a few hardcoded dimensions on this method.

Reply:
I don't like it either @3lvis , but there is a method called `calculatedGridHeaderViewFrame` in `MarketViewController` that needs it to calculate the height of the collectionView header based on whether or not the `inlineConsentView` and/or `broadcastContainerView` is included. Could probably be solved in a different way, but I did this to not have to rewrite to much and to follow the pattern the `broadcastContainerView` used when it was implemented.

Reply:
I see, nevermind then :)

Reply:
Would it be an option to not do the ` max(screenSize.width, widestButtonWidth)` and instead return the widest button width? I think this would be more "true" to the concept of intrinsic content size and instead make the client/implementation of the view handle how it should be laid out in relation to the max width. Does that make sense? 🤔😄

Reply:
Actually, I think it would be more true to return the width of the `descriptionTitleLabel` since that would in most cases be the widest of the items in the view and, in fact, be what defines the width of the view. Or am I wrong here?

Reply:
Sounds good :) Or perhaps the widest one of them? So something like `max(descriptionTitleLabelWidth, widestButtonWidth)`

Reply:
_Mirror, mirror, on the wall, who's the widest of them all?_

Reply:
😂